assets/js/blocks-registry/payment-methods/test/payment-method-config-helper.ts

```diff
@@ -47,53 +104,89 @@ describe( 'canMakePaymentWithExtensions', () => {
 		throwsCallback.mockClear();
 		falseCallback.mockClear();
 	} );
+	describe( 'getCanMakePayment', () => {
+		it( 'returns callback canMakePaymentWithFeaturesCheck if no extension callback is detected', () => {
+			const args = {
+				canMakePayment: jest.fn().mockImplementation( () => true ),
+				features: [ 'products' ],
+				paymentMethodName: 'missing-payment-method',
+			};
 
-	it( "Returns false without executing the registered callbacks, if the payment method's canMakePayment callback returns false.", () => {
-		const canMakePayment = () => false;
-		const canMakePaymentWithExtensionsResult = canMakePaymentWithExtensions(
-			canMakePayment,
-			canMakePaymentExtensionsCallbacks,
-			'cod'
-		)();
-		expect( canMakePaymentWithExtensionsResult ).toBe( false );
-		expect( trueCallback ).not.toHaveBeenCalled();
-	} );
+			const canMakePayment = helpers.getCanMakePayment(
+				args.canMakePayment,
+				args.features,
+				args.paymentMethodName
+			)( canMakePaymentArgument );
+			expect( canMakePayment ).toEqual( args.canMakePayment() );
+		} );
+		it( 'returns callbacks from the extensions when they are defined', () => {
+			const args = {
+				canMakePaymentConfiguration: jest
+					.fn()
+					.mockImplementation( () => true ),
+				features: [ 'products' ],
+				paymentMethodName: 'bacs',
+			};
 
-	it( 'Returns early when a registered callback returns false, without executing all the registered callbacks', () => {
-		canMakePaymentWithExtensions(
-			() => true,
-			canMakePaymentExtensionsCallbacks,
-			'bacs'
-		)();
-		expect( falseCallback ).toHaveBeenCalledTimes( 1 );
+			const canMakePayment = helpers.getCanMakePayment(
+				args.canMakePaymentConfiguration,
+				args.features,
+				args.paymentMethodName
+			)( canMakePaymentArgument );
+			expect( canMakePayment ).toBe( falseCallback() );
```

Comment:
Doesn't `falseCallback()` evaluate to simply `false`, so you're testing here `expect( canMakePayment ).toBe( false );` or am I misunderstanding?

Reply:
I've updated the test in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4774/commits/630a15ed3e465a247f9db92b7c1bb8d9049d0e37, hope this makes it more clear